src/Files.App/Views/MainPage.xaml.cs

```diff
@@ -167,6 +167,9 @@ public async void TabItemContent_ContentChanged(object? sender, CustomTabViewIte
 			LoadPaneChanged();
 			UpdateNavToolbarProperties();
 			await NavigationHelpers.UpdateInstancePropertiesAsync(paneArgs);
+
+			// Save the updated tab list
+			AppLifecycleHelper.SaveSessionTabs();
```

Comment:
For some time this code runs twice when a second pane is added

Reply:
This can be avoided by modifying `PangeHolderPage.IsRightPaneVisible` setter as follows:
```csharp
			set
			{
				if (value != _IsRightPaneVisible)
				{
					_IsRightPaneVisible = value;
					if (!_IsRightPaneVisible)
					{
						ActivePane = PaneLeft;
						Pane_ContentChanged(null, null);
					}

					NotifyPropertyChanged(nameof(IsRightPaneVisible));
					NotifyPropertyChanged(nameof(IsMultiPaneActive));
				}
			}
```

Reply:
Thanks! I pushed a commit with this change